scripts/install.sh

```diff
@@ -69,31 +77,107 @@ printf "Network: $NETWORK\n"
 
 cd $KEEP_BEACON_SOL_PATH
 
-printf "${LOG_START}Installing YARN dependencies...${LOG_END}"
-yarn install
+printf "${LOG_START}Installing beacon YARN dependencies...${LOG_END}"
+yarn
 
 if [ "$NETWORK" == "development" ]; then
     printf "${LOG_START}Unlocking ethereum accounts...${LOG_END}"
     KEEP_ETHEREUM_PASSWORD=$KEEP_ETHEREUM_PASSWORD \
         npx hardhat unlock-accounts --network $NETWORK
 fi
 
-printf "${LOG_START}Building contracts...${LOG_END}"
-yarn clean
-yarn build
+if [ "$SKIP_DEPLOYMENT" != true ]; then
 
-if [ "$SKIP_DEPLOYMENT" = false ] ; then
-  printf "${LOG_START}Deploying contracts...${LOG_END}"
+  # create tmp/ dir for fresh installations
+  rm -rf $TMP && mkdir $TMP && cd $TMP
 
-  npx hardhat deploy --reset --export export.json --network $NETWORK
-fi
+  # clone solidity-contracts as a dependency for beacon, ecdsa and tbtc
+  git clone https://github.com/threshold-network/solidity-contracts.git
+  cd solidity-contracts
+
+  printf "${LOG_START}Building solidity-contracts...${LOG_END}"
+  yarn && yarn clean && yarn build
+
+  # deploy solidity-contracts
+  printf "${LOG_START}Deploying solidity-contracts contracts...${LOG_END}"
+  yarn deploy --reset --network $NETWORK
+
+  yarn link
+  # create export folder
+  yarn prepack
+
+  cd $KEEP_BEACON_SOL_PATH
+
```

Comment:
Aren't we missing `yarn install` for the beacon?

Reply:
no, we do it before unlocking ethereum accounts